### PR TITLE
go-selinux: some cleanup/refactor and improve test coverage

### PR DESCRIPTION
**Pull request by thaJeztah**
See individual commits for details

### CODE REVIEW

File: go-selinux/selinux_linux_test.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func TestSetFileLabel(t *testing.T) {
 	out.Close()
 	defer os.Remove(tmp)
 
-	if err := SetFileLabel(tmp, con); err != nil {
+	err = SetFileLabel(tmp, con)
+	if err != nil {
```
Comment (kolyshkin) [inline go-selinux/selinux_linux_test.go:29]: I don't understand the reason for this change.

File: go-selinux/selinux_linux_test.go [hunk 2]
```diff
@@ -514,7 +515,8 @@ func BenchmarkChcon(b *testing.B) {
 	b.Logf("Chcon(%q, %q)", dir, con)
 	b.ResetTimer()
 	for n := 0; n < b.N; n++ {
-		if err := Chcon(dir, con, true); err != nil {
+		err = Chcon(dir, con, true)
+		if err != nil {
```
Comment (kolyshkin) [inline go-selinux/selinux_linux_test.go:519]: ditto

File: go-selinux/selinux_linux.go [hunk 3]
```diff
@@ -549,30 +549,30 @@ func (l *level) parseLevel(levelStr string) error {
 
 // rangeStrToMLSRange marshals a string representation of a range.
 func rangeStrToMLSRange(rangeStr string) (*mlsRange, error) {
-	mlsRange := &mlsRange{}
+	mlsrange := &mlsRange{}
```
Comment (kolyshkin) [inline go-selinux/selinux_linux.go:552]: can we just use `range` (or even `r`) here?

File: go-selinux/selinux_linux.go [hunk 4]
```diff
@@ -712,11 +712,11 @@ func readWriteCon(fpath string, val string) (string, error) {
 
 // peerLabel retrieves the label of the client on the other side of a socket
 func peerLabel(fd uintptr) (string, error) {
-	label, err := unix.GetsockoptString(int(fd), unix.SOL_SOCKET, unix.SO_PEERSEC)
+	lbl, err := unix.GetsockoptString(int(fd), unix.SOL_SOCKET, unix.SO_PEERSEC)
```
Comment (kolyshkin) [inline go-selinux/selinux_linux.go:715]: perhaps just `l`?

File: go-selinux/selinux_linux.go [hunk 5]
```diff
@@ -734,8 +734,8 @@ func setKeyLabel(label string) error {
 
 // get returns the Context as a string
 func (c Context) get() string {
-	if level := c["level"]; level != "" {
-		return c["user"] + ":" + c["role"] + ":" + c["type"] + ":" + level
+	if lvl := c["level"]; lvl != "" {
+		return c["user"] + ":" + c["role"] + ":" + c["type"] + ":" + lvl
```
Comment (kolyshkin) [inline go-selinux/selinux_linux.go:738]: perhaps just `l`?

File: go-selinux/selinux_linux_test.go (hunk 1 above)
Comment (thaJeztah) [reply go-selinux/selinux_linux_test.go:29]: not a huge one; my editor's linter warns about variables that are shadowed; in this case an `err` already existed. I could just remove the `:`, but thought it's more clear to break it in two lines if we're reusing the variable.

File: go-selinux/selinux_linux.go (hunk 3 above)
Comment (thaJeztah) [reply go-selinux/selinux_linux.go:552]: `range` is a reserved keyboard. Can use `r` if we think it's clear enough

File: go-selinux/selinux_linux.go [hunk 6]
```diff
@@ -606,17 +606,17 @@ func bitsetToStr(c *big.Int) string {
 	return str
 }
 
-func (l1 *level) equal(l2 *level) bool {
```
Comment (kolyshkin) [inline go-selinux/selinux_linux.go:619]: I like `l1` and `l2` more in this case, since we compare them. A linter is not always right.

**Review by kolyshkin — CHANGES_REQUESTED** [review]
Commits 1 and 4 LGTM, left a few comments for 2 and 3.

File: go-selinux/selinux_linux_test.go (hunk 1 above)
Comment (kolyshkin) [reply go-selinux/selinux_linux_test.go:29]: So, in this case the linter warning should be ignored. Yes, we have a variable named `err` already, and yes, here we create another `err` which shadows the original one. No, there is no issue here.

Or maybe just remove the `:`.

**Comment by kolyshkin** [comment]
One thing that bothers me if those linter warnings you fix here will resurface after some time, as apparently we don't have those linters enabled in CI. So, if there is a linter in golangci-lint that warns of these, the PR would make more sense.

**Comment by thaJeztah** [comment]
@kolyshkin updated; moved some of the changes to https://github.com/opencontainers/selinux/pull/191, which enables more linters (some of which catching the shadowing issues, although I haven't enabled "strict" mode, which seems to closer match what my IDE shows)

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**